crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -130,54 +128,62 @@ impl Consensus {
                         txid: transaction.compute_txid(),
                         error: err,
-                    }
-                });
-                continue;
+                    })?;
+                    continue;
+                } else {
+                    return Err(BlockValidationErrors::FirstTxIsnNotCoinbase.into());
+                }
             }
             // Amount of all outputs
-            let mut output_value = 0;
+            let mut out_value = 0;
             for output in transaction.output.iter() {
-                Self::get_out_value(output, &mut output_value).map_err(|err| TransactionError {
-                    txid: transaction.compute_txid(),
-                    error: err,
-                });
-                Self::validate_script_size(&output.script_pubkey).map_err(|err| TransactionError {
-                    txid: transaction.compute_txid(),
-                    error: err,
-                });
+                out_value += output.value.to_sat();
+
+                Self::validate_script_size(&output.script_pubkey, false).map_err(|err| {
+                    TransactionError {
+                        txid: transaction.compute_txid(),
+                        error: err,
+                    }
+                })?;
             }
             // Amount of all inputs
             let mut in_value = 0;
             for input in transaction.input.iter() {
-                Self::consume_utxos(input, &mut utxos, &mut in_value).map_err(|err| {
+                let txo = Self::get_utxo(input, &utxos).map_err(|err| TransactionError {
+                    txid: transaction.compute_txid(),
+                    error: err,
+                })?;
+
+                in_value += txo.value.to_sat();
+                // We need to know if this input spends a taproot output
+                let taproot_spend = txo.script_pubkey.is_p2tr();
+
+                Self::validate_script_size(&input.script_sig, taproot_spend).map_err(|err| {
                     TransactionError {
                         txid: transaction.compute_txid(),
                         error: err,
                     }
-                });
-                Self::validate_script_size(&input.script_sig).map_err(|err| TransactionError {
-                    txid: transaction.compute_txid(),
-                    error: err,
-                });
+                })?;
             }
             // Value in should be greater or equal to value out. Otherwise, inflation.
-            if output_value > in_value {
+            if out_value > in_value {
                 return Err(TransactionError {
                     txid: transaction.compute_txid(),
                     error: BlockValidationErrors::NotEnoughMoney,
                 }
                 .into());
             }
-            if output_value > 21_000_000 * 100_000_000 {
+            // Sanity check
+            if out_value > 21_000_000 * 100_000_000 {
```

Comment:
`21_000_000 * 100_000_000` can be a constant `TOTAL_AMOUNT_OF_SATS`  

IMO improve readability

Reply:
Yeah, actually there's a constant for the sats per BTC `COIN_VALUE`, nicer if we use it